syntax = "proto3";

// option java_multiple_files = true;
// option java_package = "io.grpc.examples.helloworld";
// option java_outer_classname = "HelloWorldProto";
// option objc_class_prefix = "HLW";

package users;

// The greeting service definition.
service Users {
    rpc GetUsers(GetUsersRequest) returns (GetUsersResponse){}
    rpc DescribeUser(DescribeUserRequest) returns (DescribeUserResponse){}
    // rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse){}
}

message User {
  string name = 1;
  string id = 2;
  string email = 3;
  int64 phonenumber = 4;
}

// The request message containing the user's name.
message GetUsersRequest {}

// The response message containing the greetings
message GetUsersResponse {
  User user = 1;
}

message DescribeUserRequest {
  User user = 1;
}

message DescribeUserResponse {
  string text = 1;
}